proposals/target_config.md

```diff
@@ -0,0 +1,158 @@
+# Meta
+[meta]: #meta
+- Name: (Target Configuration CRD for policy reporter)
+- Start Date: (2025-01-14)
+- Author(s): (aerosouund)
+
+# Table of Contents
+[table-of-contents]: #table-of-contents
+- [Meta](#meta)
+- [Table of Contents](#table-of-contents)
+- [Overview](#overview)
+- [Proposal](#proposal)
+- [Implementation](#implementation)
+- [Unresolved Questions](#unresolved-questions)
+
+# Overview
+
+The [policy reporter](https://github.com/kyverno/policy-reporter) takes in configuration for targets that it will send policy reports to through a flag `--config` which points to a file that can have a structure which is documented [here](https://kyverno.github.io/policy-reporter/core/targets). This document aims to define a CRD that will be used to provide the same functionality as this config file. giving administrators the ability to manage targets with this CRD.
+
+A single CRD equals exactly one target. For example if you want to send to multiple slack channels, you need a CRD per channel
+
+
+# Proposal
+
+The CRD is proposed to have the following structure. The resource is non namespaced, and by default applies to all existing namespaces.
+
+```yaml
+apiVersion: policyreporter.io/v1alpha1
+kind: TargetConfig
+metadata:
+  name: test
+status:
+ sendStatus:
+ - reportName: 179db659-a29f-42c5-a9c2-4106d01ea339
```

Comment:
its not always the case that a send relates to a single report. This is only the case if the target supports batch send and the policy report only relates to a single resource. Keep in mind that policy reporter also supports different tools which has different approaches.

Reply:
You could consider to create Kubernetes Events instead.

Reply:
test

Reply:
I mean you create an Kubernetes Event resource when a notification was send or failed instead of updating the status for the target.